src/data_buildblock/randoms_from_singles.cxx

```diff
@@ -30,8 +30,11 @@
 START_NAMESPACE_STIR
 
 void randoms_from_singles(ProjData& proj_data, const SinglesRates& singles,
-                          const float coincidence_time_window)
+                          const float coincidence_time_window, float isotope_halflife=-1.F)
 {
+  if (isotope_halflife == -1.F)
+    isotope_halflife = proj_data.get_exam_info().get_radionuclide().get_half_life();
```

Comment:
This relies on the `projdata` halflife being set but will error if not.

Reply:
what do you mean with "will error". Throw an error? If so, ok. But if you mean "silently return the wrong results", then that'd be a problem. I then suggest to add a check if it's still <0, then use F-18 (and write the warning).

Reply:
It is okay as is, will call the `error()` function if `check==true`, which it does by default. https://github.com/UCL/STIR/blob/07aad4371043136f6d2f0ee63893cfabb6e5104b/src/buildblock/Radionuclide.cxx#L75-L80